### DISCUSSION TITLE
how to add more dimensions to all metrics with custom extension?

### DISCUSSION THREAD

**Discussion by tuhao1020** [open-telemetry/opentelemetry-java-instrumentation#7765] [Q&A] [answered]
I want to add some environment variables as dimensions to all metrics of a service, how can I do that with custom extensions?

**Comment by mateuszrzeszutek** [comment, marked as answer]
If you're using environment variables (which are constant for a single execution of a program), how about adding them to the [resource attributes](https://github.com/open-telemetry/opentelemetry-java/blob/main/sdk-extensions/autoconfigure-spi/src/main/java/io/opentelemetry/sdk/autoconfigure/spi/ResourceProvider.java)? Resource attributes are attached to all telemetry emitted by the agent, metrics included.

**Reply by tuhao1020** [reply]
Thanks! I found that the [transform processor](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/processor/transformprocessor) in the opentelemetry collector meets my requirements
